packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2702,35 +2703,38 @@ export class ContainerRuntime
 		if (hasModernRuntimeMessageEnvelope) {
 			// If the message has the modern message envelope, then process it here.
 			// Here we unpack the message (decompress, unchunk, and/or ungroup) into a batch of messages with ContainerMessageType
-			const inboundBatch = this.remoteMessageProcessor.process(messageCopy, logLegacyCase);
-			if (inboundBatch === undefined) {
+			const inboundResult = this.remoteMessageProcessor.process(messageCopy, logLegacyCase);
+			if (inboundResult === undefined) {
 				// This means the incoming message is an incomplete part of a message or batch
 				// and we need to process more messages before the rest of the system can understand it.
 				return;
 			}
 
-			const result = this.duplicateBatchDetector.processInboundBatch(inboundBatch);
-			if (result.duplicate) {
-				const error = new DataCorruptionError(
-					"Duplicate batch - The same batch was sequenced twice",
-					{ batchId: inboundBatch.batchId },
-				);
+			if ("batchStart" in inboundResult) {
```

Comment:
nit: Is there a better way to execute this `if(...)` block other than checking if a property is present? Maybe looking at the `type` property?

Reply:
I personally like checking for presence of the property, TypeScript then narrows to either of the two types that has it.